qemu/tests/virtio_driver_sign_check.py

```diff
@@ -1,87 +1,153 @@
     session = vm.wait_for_login(timeout=timeout)
-    signtool_install = params.get("signtool_install")
-    au3_link = params.get("winsdk_au3", "autoit/winsdk.au3")
-    au3_link = os.path.join(test.bindir, au3_link)
     signtool_cmd = params.get("signtool_cmd")
     list_files_cmd = params.get("list_files_cmd")
-    drive_list = params.get("drive_list")
-    vm.copy_files_to(au3_link, "c:\\")
-    drivers = {}
-    logging.info("Install sdk in guest if signtool is not available.")
-    session.cmd_status_output(signtool_install, timeout=360)
-    # Wait until guest start install sdk.
-    time.sleep(10)
-    logging.info("Waiting for guest sdk setup ...")
-    utils_misc.wait_for(is_sdksetup_finished, timeout=1800)
-    results_all = """All the signature check log:\n"""
+    drive_volume_list = params.objects("drive_volume")
     fails_log = """Failed signature check log:\n"""
+    results_all = """All the signature check log:\n"""
     fail_num = 0
     fail_drivers = []
-    logging.info("Running SignTool command in guest...")
+    virtio_drivers = []
+    drivers = {}
+    need_test_driver = params.objects("tested_drivers")
+    get_disk_letter = params.get("get_disk_letter_cmd")
+
+    def get_os_arch_path(alias_map):
+        """
+        Get the os path and arch path which would be tested.
+
+        :param session: VM session.
+        :param alias_map: all guest_name and os path + arch path mapping info.
+
+        :return os_arch_path_list: Return the os_arch_path_list.
+        """
+        guest_name = params["guest_name"]
+        guest_list = dict([x.split(":") for x in alias_map.split(",")])
+        os_arch_path = guest_list[guest_name]
+        global os_arch_path_list
+        os_arch_path_list = os_arch_path.split("\\")
+        return os_arch_path_list
+
+    def filter_all_tested_cat_files(cat_files):
+        """
+        In all suffix '.cat' files, select that will be tested '.cat' file.
+
+        :param cat_files:all suffix '.cat' files.
+        :return True:correct os path, arch path and the dirver_dir
+        belong in need_test_driver list, return True.
+        """
+        pattern = r'(\w:)\\(.+?)\\(.+?)\\(.+?)[\.|\\]'
+        ret = re.match(pattern, cat_files, re.I)
+        disk_name = ret.group(1)
+        os_name = ret.group(3)
+
+        if disk_name == "A:":
+            get_os_arch_path(alias_map=params.get("guest_alias_fl"))
+            os_path = os_arch_path_list[1]
+            arch_path = os_arch_path_list[0]
+            plat_name = ret.group(2)
+            driver_dir = ret.group(4)
+        else:
+            get_os_arch_path(alias_map=params.get("guest_alias"))
+            os_path = os_arch_path_list[0]
+            arch_path = os_arch_path_list[1]
+            driver_dir = ret.group(2)
+            plat_name = ret.group(4)
+
+        if (os_name == os_path) and (plat_name == arch_path):
+            if driver_dir in need_test_driver:
+                return True
```

Comment:
Hi hachen, 

this value will be used at cat_list = filter(filter_all_tested_cat_files, drivers['.cat']), if driver name match filter_all_tested_cat_files judge condition also means the def filter_all_tested_cat_files return True, it will be kept, if not, it will be filtered. that's why I return True on this def define.